src/bin/hal/cmd/psbt.rs

```diff
@@ -483,37 +486,30 @@ fn exec_rawsign<'a>(args: &clap::ArgMatches<'a>) {
 	}
 
 	let tx =  psbt.clone().extract_tx();
-	let mut cache = bitcoin::util::sighash::SighashCache::new(&tx);
+	let mut cache = bitcoin::sighash::SighashCache::new(&tx);
 
-	let pk = secp256k1::PublicKey::from_secret_key(&SECP, &sk);
+	let pk = secp256k1::PublicKey::from_secret_key(&SECP, &sk.inner);
 	let pk = bitcoin::PublicKey {
 		compressed: compressed,
 		inner: pk,
 	};
+	let mut key_map = HashMap::new();
+	key_map.insert(pk, sk);
 	let msg = psbt.sighash_msg(i, &mut cache, None)
 		.need("error computing sighash message on psbt");
-	let secp_sig = match msg {
-		miniscript::psbt::PsbtSighashMsg::EcdsaSighash(sighash) => {
-			let msg = secp256k1::Message::from_slice(&sighash)
-				.need("error computing sighash message on psbt");
-			SECP.sign_ecdsa(&msg, &sk)
+	match msg {
+		PsbtSighashMsg::LegacySighash(_sighash) => {
+			psbt.sign(&key_map, &SECP).expect("ecdsa psbt signing error");
 		},
-		miniscript::psbt::PsbtSighashMsg::TapSighash(_) => {
-			panic!("Signing taproot transactions is not yet suppported")
+		PsbtSighashMsg::SegwitV0Sighash(_sighash) => {
+			psbt.sign(&key_map, &SECP).expect("ecdsa psbt signing error");
+		},
+		PsbtSighashMsg::TapSighash(_tap_sighash) => {
+			todo!("Signing taproot transactions not yet supported")
```

Comment:
I think we can use our own `exit!()` macro here that does a more graceful exit. Maybe accompanied with a todo comment?